### PR TITLE
Implement a function collecting intervals between stack samples

### PR DESCRIPTION
**Pull request by akopich**
Bug: http://b/240253740
Test: Unit

### CODE REVIEW

File: src/MizarData/include/MizarData/GetCallstackSamplingIntervals.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#include <vector>
+
+#include "ClientData/CallstackData.h"
+#include "MizarBase/ThreadId.h"
+
+namespace orbit_mizar_data {
+
+// Returns a collection of time intervals in nanos between consecutive callstack samples of the same
+// thread
+[[nodiscard]] std::vector<uint64_t> GetSamplingIntervalsNs(
```
Comment (florian-kuebler) [inline src/MizarData/include/MizarData/GetCallstackSamplingIntervals.h:19]: Do we want a Duration StrongTypedef?

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

File: src/MizarData/include/MizarData/GetCallstackSamplingIntervals.h (hunk 1 above)
Comment (akopich) [reply src/MizarData/include/MizarData/GetCallstackSamplingIntervals.h:19]: We do. And it should be actually based on std::chrono. In fact, the whole way Mizar (and Orbit actually) treats time has to be reworked. But that's not a subject of this PR.

File: src/MizarData/GetCallstackSamplingIntervalsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    EXPECT_THAT(GetActualIntervals({kFirstTID, kSecondTID}), IsEmpty());
+    EXPECT_THAT(GetActualIntervals({kFirstTID}), IsEmpty());
+    EXPECT_THAT(GetActualIntervals({kSecondTID}), IsEmpty());
+    EXPECT_THAT(GetActualIntervals({}), IsEmpty());
+  }
+
+  std::unique_ptr<CallstackData> callstack_data_ = std::make_unique<CallstackData>();
+};
+
+TEST_F(GetCallstackSamplingIntervalsTest, IsCorrectForNoEvents) {
```
Comment (vwbaker) [inline src/MizarData/GetCallstackSamplingIntervalsTest.cpp:75]: nit; if this is common in Orbit then feel free to ignore, but the "IsCorrectFor" seems redundant to me, especially for a function with no failure cases.  I would opt for things like "ReturnsEmptyForNoEvents" and for the more complicated ones, either "TestSingleEventInOneThreadAndMultipleInTheOther" or just "SingleEventInOneThreadAndMultipleInTheOther"

https://engdoc.corp.google.com/eng/doc/devguide/cpp/cpp101/codelab.md#test-suites

**Review by vwbaker — APPROVED** [review]

**Comment by vwbaker** [comment]
LGTM. In general, I prefer 1-sentence or something to describe what this will be used for in the commit message though.

File: src/MizarData/GetCallstackSamplingIntervalsTest.cpp (hunk 2 above)
Comment (akopich) [reply src/MizarData/GetCallstackSamplingIntervalsTest.cpp:75]: You're right, thank you.

**Pull request merged**